systemtest/src/main/java/hivemall/systemtest/runner/SystemTestRunner.java

```diff
@@ -0,0 +1,335 @@
+
+            if (q instanceof CreateTableHQ) {
+                // memo table initialized each class as immutable
+                immutableTables.add(((CreateTableHQ) q).tableName);
+            } else if (q instanceof UploadFileAsNewTableHQ) {
+                immutableTables.add(((UploadFileAsNewTableHQ) q).tableName);
+            }
+        }
+    }
+
+    // drop temporary database (should be called per Test class)
+    void resetDB() throws Exception {
+        dropDB(dbName);
+    }
+
+    public final boolean isImmutableTable(final String tableName) {
+        return immutableTables.contains(tableName);
+    }
+
+    // execute HQBase
+    public List<String> exec(@Nonnull final HQBase hq) throws Exception {
+        if (hq instanceof RawHQ) {
+            return exec((RawHQ) hq);
+        } else if (hq instanceof TableHQ) {
+            return exec((TableHQ) hq);
+        } else if (hq instanceof TableListHQ) {
+            return tableList();
+        } else {
+            throw new IllegalArgumentException("Unexpected query type: " + hq.getClass());
+        }
+    }
+
+    //// execute RawHQ
+    abstract protected List<String> exec(@Nonnull final RawHQ hq) throws Exception;
+
+    //// execute TableHQ
+    List<String> exec(@Nonnull final TableHQ hq) throws Exception {
+        if (hq instanceof CreateTableHQ) {
+            return createTable((CreateTableHQ) hq);
+        } else if (hq instanceof DropTableHQ) {
+            return dropTable((DropTableHQ) hq);
+        } else if (hq instanceof InsertHQ) {
+            return insert((InsertHQ) hq);
+        } else if (hq instanceof UploadFileHQ) {
+            return exec((UploadFileHQ) hq);
+        } else {
+            throw new IllegalArgumentException("Unexpected query type: " + hq.getClass());
+        }
+    }
+
+    ////// execute UploadFileHQ
+    List<String> exec(@Nonnull final UploadFileHQ hq) throws Exception {
+        if (hq instanceof UploadFileAsNewTableHQ) {
+            return uploadFileAsNewTable((UploadFileAsNewTableHQ) hq);
+        } else if (hq instanceof UploadFileToExistingHQ) {
+            return uploadFileToExisting((UploadFileToExistingHQ) hq);
+        } else {
+            throw new IllegalArgumentException("Unexpected query type: " + hq.getClass());
+        }
+    }
+
+    // matching HQBase
+    public void matching(@Nonnull final HQBase hq, @CheckForNull final String answer,
+            final boolean ordered) throws Exception {
+        Preconditions.checkNotNull(answer);
+
+        List<String> result = exec(hq);
+
+        if (ordered) {
+            // take order into consideration (like list)
+            Assert.assertThat(Arrays.asList(answer.split(IO.RD)),
+                Matchers.contains(result.toArray()));
+        } else {
+            // not take order into consideration (like multiset)
+            Assert.assertThat(Arrays.asList(answer.split(IO.RD)),
+                Matchers.containsInAnyOrder(result.toArray()));
```

Comment:
@Lewuathe I cannot reproduce it. plz check that `answer/test3` uses tab characters as delimiter, since several space characters have got mixed in README(will be fixed)